pkg/zonemgr/zone_file.go

```diff
@@ -2,180 +2,144 @@ package zonemgr
 
 import (
 	"fmt"
-	v1 "kubevirt.io/api/core/v1"
+	"reflect"
+	"sort"
 	"strconv"
+
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/net"
+
+	v1 "kubevirt.io/api/core/v1"
 )
 
 const (
 	refresh = "3600"    // 1 hour (seconds) - how long a nameserver should wait prior to checking for a Serial Number increase within the primary zone file
 	retry   = "3600"    // 1 hour (seconds) - how long a nameserver should wait prior to retrying to update a zone after a failed attempt.
 	expire  = "1209600" // 2 weeks (seconds) - how long a nameserver should wait prior to considering data from a secondary zone invalid and stop answering queries for that zone
 	ttl     = "3600"    // 1 hour (seconds) - the duration that the record may be cached by any resolver
+
+	domainDefault     = "vm"
+	nameServerDefault = "ns"
+	adminEmailDefault = "email"
 )
 
 type ZoneFile struct {
-	soaSerial          int
-	clusterName        string
-	userAdminEmail     string
-	userNameserverName string
-	userNameserverIP   string
-	userSubdomain      string
-
-	domainName string
+	soaSerial      int
+	adminEmail     string
+	nameServerName string
+	nameServerIP   string
+	domain         string
+
 	headerPref string
 	headerSuf  string
 
 	header   string
 	aRecords string
 	content  string
 
-	vmiRecordsMap map[string]map[string]ARecord // key=vmi_name+namespace, value=map[key=iface_name, value=ip]
-	//ifacesMap map[string]string
+	vmiRecordsMap map[string][]string
 }
 
-func NewZoneFile(clusterName string, userAdminEmail string, userNameserverName string,
-	userNameserverIP string, userSubdomain string) *ZoneFile {
+func NewZoneFile(nameServerIP string, domain string) *ZoneFile {
```

Comment:
You have `NewZoneFile` and` NewDnsZoneFile`. I find the names confusing. What is the purpose of each one them them?

Reply:
DnsZoneFile refers to disk zone file. ZoneFile refers to cached zone file.

Reply:
yes, better ZoneFile and ZoneFileCache.